sklearn/feature_selection/from_model.py

```diff
@@ -17,11 +18,21 @@ def _get_feature_importances(estimator):
     importances = getattr(estimator, "feature_importances_", None)
 
     if importances is None and hasattr(estimator, "coef_"):
-        if estimator.coef_.ndim == 1:
-            importances = np.abs(estimator.coef_)
+        if isinstance(estimator.coef_, np.ndarray):
+            if estimator.coef_.ndim == 1:
+                importances = np.abs(estimator.coef_)
+            else:
+                importances = np.sum(np.abs(estimator.coef_), axis=0)
+
+        elif sps.isspmatrix_csr(estimator.coef_):
+            importances = np.sum(np.abs(estimator.coef_.toarray()), axis=0)
```

Comment:
So far as I understand, `np.sum` returns a numpy array so the type of `importances` should be ok. I used `toarray()` on `estimater.coef_` to convert it from `csr_matrix`. The scipy.sparse documentation strongly recommended against using numpy functions to operate on `csr_matrix`.

http://docs.scipy.org/doc/scipy/reference/sparse.html